src/constraints/transport.jl

```diff
@@ -6,30 +6,22 @@ add_transport_constraints!(model, graph, df_flows, flow, Ft, flows_investment)
 Adds the transport flow constraints to the model.
 """
 
-function add_transport_constraints!(model, graph, df_flows, flow, Ft, flows_investment)
+function add_transport_constraints!(
+    model,
+    graph,
+    df_flows,
+    flow,
+    Ft,
+    accumulated_flows_export_units,
+    accumulated_flows_import_units,
+    flows_investment,
+)
 
     ## Expressions used by transport flow constraints
+
     # - Create upper limit of transport flow
     upper_bound_transport_flow = [
-        if graph[row.from, row.to].investable[row.year]
-            @expression(
-                model,
-                profile_aggregation(
-                    Statistics.mean,
-                    graph[row.from, row.to].rep_periods_profiles,
-                    row.year,
-                    row.year,
-                    ("availability", row.rep_period),
-                    row.timesteps_block,
-                    1.0,
-                ) *
-                graph[row.from, row.to].capacity *
-                (
-                    graph[row.from, row.to].initial_export_units[row.year] +
-                    flows_investment[row.year, (row.from, row.to)]
-                )
-            )
-        else
+        if (row.from, row.to) in Ft
```

Comment:
With this change `upper_bound_transport_flow` and `lower_...` now have a bunch of empty because the `if` doesn't have an else.
I still have to understand where this is used, but in principle you could change the `for row in ...` to add a `if (row.from, row.to) in Ft` at the end. This will make the `upper_....` vector limited to that size only.

Reply:
So it looks like this expression has length `length(df_flows)`, which means that if we change the loop to `for row in .... if ... in Ft`, the size will be wrong.

However, in the next section you define `df` which is exactly the filter that we want.

So it looks like you could move the `df` computation to above this and redefine this expression as a loop over the rows of `df`.

Later, when you access this expression via `row.index`, it will break because the index is not correct anymore.
So, instead of that index, we actually want the indexes of `df`. So we can change to something like

```Julia
...
   this_expression[idx]
...
   for (idx, row) in enumerate(eachrow(df))
```

Alternatively, you can `zip` and loop over the values of both:

```Julia
...
    element
...
    for (row, element) in zip(eachrow(df), this_expression)
```

I hope this wasn't too confusing

Reply:
Thanks! I think I got it!